Tests/iaas/flavor-naming/flavor_names.py

```diff
@@ -234,6 +239,11 @@ def shorten(self):
         """return canonically shortened name as recommended in the standard"""
         if self.hype is None and self.hwvirt is None and self.cpubrand is None:
             return self
+        # For non-x86-64, don't strip out CPU brand for short name, as it contains the architecture
+        if self.cpubrand and self.cpubrand.cpuvendor != 'i' and self.cpubrand.cpuvendor != 'z':
```

Comment:
I still prefer 

```suggestion
        if self.cpubrand and self.cpubrand.cpuvendor not in ('i', 'z'):
```

I would also have preferred the variant with the single call of the constructor, but that's mainly my taste.